Support empty nodeAccountId in TransactionBody

**Detailed Description**
Services made a change in https://github.com/hashgraph/hedera-services/pull/1138 that allows `TransactionBody.nodeAccountId` to be null for executed scheduled transactions.

**Actual Behavior**
Steps to reproduce the behavior:
1. Parse a record file with an executed scheduled transaction
```
Mar 08 15:06:59 mirrornode-preview-testnet-parser-1 java[2542]: 2021-03-08T09:06:59.948-0600 ERROR scheduling-5 c.h.m.i.p.r.RecordFileParser Error parsing file 2021-03-05T21_10_34.922961000Z.rcd com.hedera.mirror.importer.exception.ParserException: Error copying 4 items to table transaction\
        at com.hedera.mirror.importer.parser.record.entity.sql.PgCopy.copy(PgCopy.java:99)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.executeBatches(SqlEntityListener.java:189)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:157)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:67)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:165)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$FastClassBySpringCGLIB$$8b9310db.invoke(<generated>)\
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)\
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:367)\
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:118)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:91)\
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:287)\
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:164)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:118)\
        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:153)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)\
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$7f5a4869.parse(<generated>)\
        at jdk.internal.reflect.GeneratedMethodAccessor142.invoke(Unknown Source)\
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper$HandlerMethod.invoke(MessagingMethodInvokerHelper.java:1104)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.invokeHandlerMethod(MessagingMethodInvokerHelper.java:582)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.processInternal(MessagingMethodInvokerHelper.java:477)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.process(MessagingMethodInvokerHelper.java:355)\
        at org.springframework.integration.handler.MethodInvokingMessageProcessor.processMessage(MethodInvokingMessageProcessor.java:108)\
        at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:104)\
        at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:134)\
        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:62)\
        at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:444)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:428)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:376)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$null$3(AbstractPollingEndpoint.java:323)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)\
        at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$4(AbstractPollingEndpoint.java:320)\
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)\
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:93)\
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\
        at java.base/java.lang.Thread.run(Thread.java:834)\
Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "node_account_id" violates not-null constraint\
  Detail: Failing row contains (1614978635553604000, 27, 22, 1783, 1614978624410376201, 120, null, 18474, 0, 200000000, 12331, \x, \xa47a4cdb731f5ed435dd8289af27f89df80b8b5ef19c490b21454da379579b..., null, t).\
  Where: COPY transaction, line 4: "12331,1614978635553604000,18474,0,200000000,"\x",,1783,22,true,,"\xa47a4cdb731f5ed435dd8289af27f89df..."\
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)\
        at org.postgresql.core.v3.QueryExecutorImpl.processCopyResults(QueryExecutorImpl.java:1212)\
        at org.postgresql.core.v3.QueryExecutorImpl.endCopy(QueryExecutorImpl.java:1017)\
        at org.postgresql.core.v3.CopyInImpl.endCopy(CopyInImpl.java:49)\
        at org.postgresql.copy.CopyManager.copyIn(CopyManager.java:185)\
        at com.hedera.mirror.importer.parser.record.entity.sql.PgCopy.copy(PgCopy.java:95)\
        ... 52 more
Mar 08 15:06:59 mirrornode-preview-testnet-parser-1 java[2542]: 2021-03-08T09:06:59.948-0600 INFO scheduling-5 c.h.m.i.p.r.RecordFileParser Finished parsing 4 transactions from record file 2021-03-05T21_10_34.922961000Z.rcd in 3.307 ms. Success: false
```

**Expected Behavior**
Transaction is ingested

**Environment:**
- Java: OpenJDK 11
- OS: Ubuntu 18.04
- Version: v0.28.0

**Additional Context**
Add any other context about the problem here. Attach any logs here, if applicable.

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.35.4__transaction_node_account_id_null.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.35.4__transaction_node_account_id_null.sql
@@ -1,0 +1,7 @@
+-------------------
+-- Change transaction.node_account_id to no longer be required. This change is necessary since scheduled transactions
+-- aren't associated with a particular node.
+-------------------
+
+alter table if exists transaction
+    alter column node_account_id drop not null;

--- a/hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__time_scale_init.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__time_scale_init.sql
@@ -512,7 +512,7 @@ create table if not exists transaction
     payer_account_id       bigint   not null,
     valid_start_ns         bigint   not null,
     valid_duration_seconds bigint,
-    node_account_id        bigint   not null,
+    node_account_id        bigint,
     entity_id              bigint,
     initial_balance        bigint            default 0,
     max_fee                bigint,

--- a/hedera-mirror-rest/__tests__/integrationDomainOps.js
+++ b/hedera-mirror-rest/__tests__/integrationDomainOps.js
@@ -235,7 +235,7 @@ const addTransaction = async (transaction) => {
   }
 
   const payerAccount = EntityId.fromString(transaction.payerAccountId);
-  const nodeAccount = EntityId.fromString(transaction.nodeAccountId);
+  const nodeAccount = EntityId.fromString(transaction.nodeAccountId, true);
   await sqlConnection.query(
     `INSERT INTO transaction (
       consensus_ns, valid_start_ns, payer_account_id, node_account_id,

--- a/hedera-mirror-rest/__tests__/specs/transactions-01-no-args.spec.json
+++ b/hedera-mirror-rest/__tests__/specs/transactions-01-no-args.spec.json
@@ -23,7 +23,7 @@
       {
         "charged_tx_fee": 0,
         "payerAccountId": "0.0.8",
-        "nodeAccountId": "0.0.3",
+        "nodeAccountId": null,
         "consensus_timestamp": "1234567800000000009",
         "name": "CRYPTOUPDATEACCOUNT",
         "type": "15"
@@ -245,7 +245,7 @@
         "max_fee": "33",
         "memo_base64": null,
         "name": "CRYPTOUPDATEACCOUNT",
-        "node": "0.0.3",
+        "node": null,
         "result": "SUCCESS",
         "scheduled": false,
         "transaction_hash": "aGFzaA==",

--- a/hedera-mirror-rest/transactions.js
+++ b/hedera-mirror-rest/transactions.js
@@ -106,7 +106,7 @@ const createTransferLists = (rows) => {
         max_fee: utils.getNullableNumber(row.max_fee),
         memo_base64: utils.encodeBase64(row.memo),
         name: row.name,
-        node: EntityId.fromString(row.node_account_id).toString(),
+        node: EntityId.fromString(row.node_account_id, true).toString(),
         result: row.result,
         scheduled: row.scheduled,
         token_transfers: createTokenTransferList(row.token_transfer_list),

